underfs/swift/src/main/java/tachyon/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -38,35 +74,353 @@
-    config.addResource(new Path(config.get(Constants.UNDERFS_HDFS_CONFIGURATION)));
-    super.prepareConfiguration(path, tachyonConf, config);
+  @Override
+  public long getBlockSizeByte(String path) throws IOException {
+    return Constants.GB * 4;
+  }
+
+  @Override
+  public Object getConf() {
+    LOG.warn("getConf is not supported when using SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public List<String> getFileLocations(String path) throws IOException {
+    LOG.warn("getFileLocations is not supported when using "
+        + "SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public List<String> getFileLocations(String path, long offset) throws IOException {
+    LOG.warn("getFileLocations is not supported when using "
+        + "SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public long getFileSize(String path) throws IOException {
+    StoredObject so = mAccount.getContainer(mContainerName).getObject(stripPrefixIfPresent(path));
+    return so.getContentLength();
+  }
+
+  @Override
+  public long getModificationTimeMs(String path) throws IOException {
+    StoredObject so = mAccount.getContainer(mContainerName).getObject(stripPrefixIfPresent(path));
+    return so.getLastModifiedAsDate().getTime();
+  }
+
+  // This call is currently only used for the web ui, where a negative value implies unknown.
+  @Override
+  public long getSpace(String path, SpaceType type) throws IOException {
+    return -1;
+  }
+
+  @Override
+  public boolean isFile(String path) throws IOException {
+    return exists(path);
+  }
+
+  @Override
+  public String[] list(String path) throws IOException {
+    path = path.endsWith(PATH_SEPARATOR) ? path : path + PATH_SEPARATOR;
+    return listInternal(path, false);
+  }
+
+  /* @inheritDoc
+   * @see tachyon.underfs.UnderFileSystem#mkdirs(java.lang.String, boolean)
+   * There is no notion of directories in Swift.
+   * The content of containers are objects.
+   * Object name may contain nested structure like a/b/c/d.data
+   */
+  @Override
+  public boolean mkdirs(String path, boolean createParent) throws IOException {
+    return true;
+  }
+
+  @Override
+  public InputStream open(String path) throws IOException {
+    path = stripPrefixIfPresent(path);
+    Container container = mAccount.getContainer(mContainerName);
+    StoredObject so = container.getObject(path);
+    InputStream  is = so.downloadObjectAsInputStream();
+    return is;
+  }
+
+  /**
+   * Each path is checked both for leading "/" and ending "/"
+   * Leading "/" is removed, and "/" is added at the end if not present
+   * @param path
+   * @return qualified path
+   */
+  private String makeQualifiedPath(String path) {
+    if (!path.endsWith("/")) {
+      path = path + "/";
+    }
+    if (path.startsWith("/")) {
+      path = path.substring(1);
+    }
+    return path;
+  }
+
+  /* @inheritDoc
```

Comment:
After re-reading the comment and also looking at the inherit documentation, it makes sense to have both. So this is okay as is. Thanks.